core/src/main/java/org/pentaho/reporting/platform/plugin/output/OutputUtils.java

```diff
@@ -23,21 +23,38 @@ private OutputUtils() {
 
   static final Log log = LogFactory.getLog( PDFOutput.class );
 
-  public static void overrideQueryLimit( MasterReport report ) {
-    report.setQueryLimit( getPropertyValue( "export-query-limit", report.getQueryLimit()  ) );
+  public static void enforceQueryLimit( MasterReport report ) {
+    report.setQueryLimit( getPropertyValue( "export-query-limit", report.getQueryLimit() ) );
   }
 
-  private static int getPropertyValue( String propName, int def ) {
-    String limit = PentahoSystem.getSystemSetting( "pentaho-interactive-reporting/settings.xml", propName, String.valueOf( def ) );
+  private static int getPropertyValue( String propName, int reportLimit ) {
+    String setting = PentahoSystem.getSystemSetting( "pentaho-interactive-reporting/settings.xml", propName, "-1" );
```

Comment:
What happens in that scenario were the property did not exist in "settings.xml"?